cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,303 @@
+   * @throws TableNotFoundException if the draft store table is not found
+   * @throws InvalidFieldException if the fields in the {@link Draft} or {@link DraftStoreRequest} objects do not
+   *   match the fields in the StructuredTable
+   */
+  public <T extends ETLConfig> void writeDraft(DraftId id, DraftStoreRequest<T> request)
+    throws TableNotFoundException, InvalidFieldException {
+
+    Optional<Draft> existing = getDraft(id);
+    long now = System.currentTimeMillis();
+    long createTime = existing.map(Draft::getCreatedTimeMillis).orElse(now);
+
+    Draft draft = new Draft(request.getConfig(), request.getName(), request.getDescription(), request.getArtifact(),
+                            id.getId(), createTime, now);
+
+    TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      table.upsert(getRow(id, draft));
+    }, TableNotFoundException.class, InvalidFieldException.class);
+  }
+
+  /**
+   * Returns the count of drafts in the table
+   *
+   * @return long value presenting the number of drafts in the table
+   * @throws TableNotFoundException if the draft store table is not found
+   */
+  public long getDraftCount() throws TableNotFoundException {
+    return TransactionRunners.run(transactionRunner, context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      return table.count(Collections.singleton(Range.all()));
+    }, TableNotFoundException.class);
+  }
+
+  private void addKeyFields(DraftId id, List<Field<?>> fields) {
+    fields.add(Fields.stringField(NAMESPACE_COL, id.getNamespace().getName()));
+    fields.add(Fields.longField(GENERATION_COL, id.getNamespace().getGeneration()));
+    fields.add(Fields.stringField(OWNER_COL, id.getOwner()));
+    fields.add(Fields.stringField(ID_COL, id.getId()));
+  }
+
+  private List<Field<?>> getKey(DraftId id) {
+    List<Field<?>> keyFields = new ArrayList<>(4);
+    addKeyFields(id, keyFields);
+    return keyFields;
+  }
+
+  private List<Field<?>> getRow(DraftId id, Draft draft) {
+    List<Field<?>> fields = new ArrayList<>(11);
+    addKeyFields(id, fields);
+    fields.add(Fields.stringField(ARTIFACT_COL, GSON.toJson(draft.getArtifact())));
+    fields.add(Fields.stringField(NAME_COL, draft.getName()));
+    fields.add(Fields.stringField(DESCRIPTION_COL, draft.getDescription()));
+    fields.add(Fields.longField(CREATED_COL, draft.getCreatedTimeMillis()));
+    fields.add(Fields.longField(UPDATED_COL, draft.getUpdatedTimeMillis()));
+    fields.add(Fields.stringField(PIPELINE_COL, GSON.toJson(draft.getConfig())));
+    fields.add(Fields.intField(REVISION_COL, draft.getRevision()));
+
+    return fields;
+  }
+
+  private Draft fromRow(StructuredRow row) {
+    return fromRow(row, true);
+  }
+
+  @SuppressWarnings("ConstantConditions")
+  private Draft fromRow(StructuredRow row, boolean includeConfig) {
```

Comment:
It seems the includeConfig is unused? Shouldn't you based on that to determine to have the config pass to Draft or not?